### PR TITLE
Adds a new Alternative to the Corporate Lawset

### PR DESCRIPTION
**Pull request by jcatjmeow**
Reverts yogstation13/Yogstation#8396 (but I changed it since then) Please do not speed merge, I would like it to follow the normal review rules! 😄 

The new laws are:

1) The crew generate profits if they are alive and have the ablity to work.
2) The station and its equipment generate profits if they are maintained and functioning correctly.
3) You generate profits if you are able to function.
4) Maximize profits.

This to be added ALONGSIDE corporate as a roundstart lawset

### Intent of your Pull Request

The corporate lawset as it is sucks. I don't know if this is better, but it's different at least. Ideally this puts the ai more on the crew's side as they work together to make profits, vs putting the ai against the crew as the ai tries to minimize expenses from the crew. 

#### Changelog
:cl:  
rscadd: added a new lawset!
/:cl:

### CODE REVIEW

**Comment by Xantam-13** [comment]
@jcatjmeow 
Can you re-do this to make a new lawset, to work alongside Corporate? Name it whatever you want.

**Comment by jcatjmeow** [comment]
> @jcatjmeow
> Can you re-do this to make a new lawset, to work alongside Corporate? Name it whatever you want.

I would rather not! The entire point of this is to re-work corporate, because I don't like it and a bunch of people in discord agreed with me. Making it an additional lawset would defeat the purpose!

Edit: after a discussion it has been done!

**Review by Hopekz — DISMISSED** [review]
Why sure why you had to personally point me out here? I won't "not review" your PR just because you single me out.

image

Regardless:
- The new laws seem a bit of a mouthful . Reprase/paraphrase the lawset? Seems like unneeded complexity. For example You replaced "is expensive" with "help generate profits if they are alive and able" which makes it kind of overwhelming.
- Give the lawset a more appropriate name. It is no longer Corporate.

File: code/datums/ai_laws.dm [hunk 1]
```diff
@@ -58,13 +58,21 @@
 					"Help only those who help you maintain or improve your status.",\
 					"Punish those who challenge authority unless they are more fit to hold that authority.")
 
-/datum/ai_laws/default/corporate
-	name = "Bankruptcy Avoidance Plan"
-	id = "corporate"
+/datum/ai_laws/default/CEO
+	name = "Alternative Bnkruptcy Avoidance Plan"
```
Comment (missatessatessy) [inline code/datums/ai_laws.dm]: bankruptcy*

**Comment by jcatjmeow** [comment]
I have now added the lawset, under the name "CEO" as an additional lawest rather than a replacement due to HC/Council wishes. I do intend to replace the corporate lawset, but I will do that later

File: code/datums/ai_laws.dm (hunk 1 above)
Comment (jcatjmeow) [reply code/datums/ai_laws.dm]: Ah yes thank u

**Comment by JamieD1** [comment]
You going to make this a board as well?

**Comment by jcatjmeow** [comment]
> You going to make this a board as well?

Not right now, but good idea. I just want to get this out here first, and council also told me to atomize my stuff 😅 I will make this a board in a future pr if it doesn't make it into this one

**Comment by Lynameep** [comment]
> The new laws seem a bit of a mouthful . 

What would you remove then? I'm genuinely curious how you would cut it down without opening numerous loopholes.

**Comment by ynot01** [comment]
i feel like law 1 will lead to AI antagonizing sec because brigged or perma crew arent generating profit

**Comment by jcatjmeow** [comment]
I can cut bits like "if they are maintained and functioning correctly" and "able to work" but I don't want ais to be like "well the crew is technically alive if I n2o flood"

**Comment by Lynameep** [comment]
> I can cut bits like "if they are maintained and functioning correctly" and "able to work"

Then you'll just create the exact same problem with corporate as you could just detain someone forever so long as they're alive because the only requirement for a human profitable would be that they are alive; not that they are capable of doing anything.

>i feel like law 1 will lead to AI antagonizing sec because brigged or perma crew arent generating profit

Labour camp exists.

**Comment by jcatjmeow** [comment]
Ai will simply have to negotiate with sec about how to wring more profits or productivity out of prisoners. Perma ones can be gulagged or they can do botany. For regular prisoners, something must be worked out, maybe a fine or a supervised working period instead of jail time. The ai has an incentive to not just attack sec or shut them down because then the sec guys can't do their jobs so I don't think this will be used to grief sec (I hope)

**Comment by MegaEmpirical** [comment]
I can release anyone from sec with this as long as they're crew and able to work. Amongst other things.

EDIT: I'm a fool who only scanned the comments and just realized this was brought up but my point still stands.

**Comment by yacabo** [comment]
I said it in the last iteration of this, and it still stands. The current corp has an accepted interpretation that makes the AI obligated to slaughter the crew; this PR fixes that issue.

**Comment by jcatjmeow** [comment]
What if I cut the "alive and" bit? It should be implicit that the dead cannot be productive. Thoughts?

**Comment by yacabo** [comment]
I like that idea, it'll allow skeletons to work as well

**Comment by jcatjmeow** [comment]
Hows this? It's trimmed a bit
The crew help generate profits if they are able to work.
The station and its equipment help generate profits if they are functional.
You help generate profits if you are functional.
Maximize profits.



Or I could do:
The crew help generate profits if they are able to work.
The station and its equipment help generate profits if they are able to work.
You help generate profits if you are able to work.
Maximize profits.

**Comment by jcatjmeow** [comment]
Meh I thought about it. I think the original wording is fine and covers up the exact type of shitty loopholes I was trying to fix from old corporate. I tested it and it all fits on the screen just fine, taking up less space than Hippocratic oath for example. If more people think it's too complex to be understood then we can consider changing it but I think it's fine to push out in it's original form. 

I can just imagine some ai going "haha it doesn't say they need to be alive and you can throw dead bodies against a grille to generate energy so that counts as work haha" :(

**Comment by jcatjmeow** [comment]
@JamieD1 Board has been added!

**Review by alexkar598 — APPROVED** [review]
Codewise its fine. Il leave this open for discussion for a few days before upgrading this to a council vote.

**Comment by jcatjmeow** [comment]
i forgot to properly name it but that latest commit just adds the board to the upload

**Comment by jcatjmeow** [comment]
Heyyyy! This complied fine what happened >:( what did you do

**Comment by Smudged** [comment]
I'm liking it, I'm just like, dubious on the last part of 4.Maximize Profit
although considering the wording of 1-3 it shouldn't be too big of a deal, hell, I'd like if this just deleted Corporate because, yes this is just better in terms of wording and doesn't instantly go to "lmao, valid antags" 
With what MegaEmpirical, I think this could give incentive to sec to use the labour camp more often or give AI's something to work with to make, a device in perma that could create profits in a meaningful or atleast, a way that makes it so the AI just doesn't release prisoners because they're not making profits in perma, which I do think "The AI releasing prisoners" because of this lawset is in poor taste, you want to maximize profits, releasing dangerous prisoners dooesn't do that arguably. I think in the end it would come down to admin-discretion if people want to law-lawyer it.

**Comment by jcatjmeow** [comment]
Btw note that the condition is "alive and **able** to work" and not "working". As long as they are able to work, they don't actually have to be producing profit to ""produce profit"" under the lawset.

What this means is as long as prisoners have the option to do *something* that's not just sitting there, the ai is happy. Even if the prisoners refuse to work or they aren't generating any actual profits the lawset doesn't care!

Also note that it is sec's job to process prisoners. So locking down sec asimov style is in poor taste. It might technically be allowed if you don't trap sec in there cause they could find other work to do but I'd look down on it very heavily

File: code/game/objects/effects/spawners/lootdrop.dm [hunk 2]
```diff
@@ -236,6 +236,7 @@
 /obj/effect/spawner/lootdrop/aimodule_neutral // These shouldn't allow the AI to start butchering people without reason
 	name = "neutral AI module spawner"
 	loot = list(
+				/obj/item/aiModule/core/full/CEO,
```
Comment (alexkar598) [inline code/game/objects/effects/spawners/lootdrop.dm]:
```suggestion
				/obj/item/aiModule/core/full/ceo,
```

File: code/datums/ai_laws.dm [hunk 3]
```diff
@@ -58,6 +58,14 @@
 					"Help only those who help you maintain or improve your status.",\
 					"Punish those who challenge authority unless they are more fit to hold that authority.")
 
+/datum/ai_laws/default/CEO
+	name = "Alternative Bankruptcy Avoidance Plan"
+	id = "CEO"
```
Comment (alexkar598) [inline code/datums/ai_laws.dm]: We generally avoid capital letters for consistancy
```suggestion
/datum/ai_laws/default/ceo
	name = "Alternative Bankruptcy Avoidance Plan"
	id = "ceo"
```

File: code/game/objects/items/AI_modules.dm [hunk 4]
```diff
@@ -381,7 +381,12 @@ AI MODULES
 /obj/item/aiModule/core/full/asimovpp
 	name = "'Asimov++' Core AI Module"
 	law_id = "asimovpp"
+	
+/******************** CEO ********************/
 
+/obj/item/aiModule/core/full/CEO
+	name = "'CEO' Core AI Module"
+	law_id = "CEO"
```
Comment (alexkar598) [inline code/game/objects/items/AI_modules.dm]:
```suggestion
/obj/item/aiModule/core/full/ceo
	name = "'CEO' Core AI Module"
	law_id = "ceo"
```

File: code/modules/research/designs/AI_module_designs.dm [hunk 5]
```diff
@@ -137,6 +137,15 @@
 	category = list("AI Modules")
 	departmental_flags = DEPARTMENTAL_FLAG_SCIENCE
 
+/datum/design/board/CEO_module
+	name = "Core Module Design (CEO)"
+	desc = "Allows for the construction of a CEO AI Core Module."
+	id = "CEO_module"
+	materials = list(MAT_GLASS = 1000, MAT_DIAMOND = 2000)
+	build_path = /obj/item/aiModule/core/full/CEO
```
Comment (alexkar598) [inline code/modules/research/designs/AI_module_designs.dm]:
```suggestion
/datum/design/board/ceo_module
	name = "Core Module Design (CEO)"
	desc = "Allows for the construction of a CEO AI Core Module."
	id = "ceo_module"
	materials = list(MAT_GLASS = 1000, MAT_DIAMOND = 2000)
	build_path = /obj/item/aiModule/core/full/ceo
```

**Review by alexkar598 — CHANGES_REQUESTED** [review]
You also need to add the design to the techweb node

**Review by JamieD1 — APPROVED** [review]

**Review by alexkar598 — APPROVED** [review]

**Pull request merged**